### PR TITLE
Draft: Move to upstream signature-help

### PR DESCRIPTION
**Pull request by PizieDust**
cc @voodoos @xvw

### CODE REVIEW

**Comment by voodoos** [comment]
The diff in the testsuite shows that the documentation if not fetch as it should be. We should either have Merlin fetch the doc itself or return the position of the function identifier, I will have a look.

File: ocaml-lsp-server/src/signature_help.ml [hunk 1]
```diff
@@ -195,51 +42,24 @@ let run (state : State.t) { SignatureHelpParams.textDocument = { uri }; position
...
-    let* application_signature =
-      let* inside_comment = Check_for_comments.position_in_comment ~position ~merlin in
-      match inside_comment with
-      | true -> Fiber.return None
-      | false ->
-        Document.Merlin.with_pipeline_exn ~name:"signature-help" merlin (fun pipeline ->
-          let typer = Mpipeline.typer_result pipeline in
-          let pos = Mpipeline.get_lexing_pos pipeline pos in
-          let node = Mtyper.node_at typer pos in
-          application_signature node ~prefix)
```
Comment (voodoos) [inline ocaml-lsp-server/src/signature_help.ml:220]: I thin you should be able to use `Merlin_analysis.Signature_help.application_signature` as a drop-in replacement here and keep the rest of the `run` function identical :-)

File: CHANGES.md [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 - Add a code-action for syntactic and semantic movement shortcuts based on Merlin's Jump command (#1364)
 
+- Upstream the implementation of signature help to merlin-lib (#1375)
```
Comment (rgrinberg) [inline CHANGES.md:18]: CHANGES are for the users. Don't think they're going to find this information useful.

**Review by xvw — APPROVED** [review]
👍🏾 
Since the patch depend on a release of Merlin, I took the liberty to flag-it as a draft!

**Review by rgrinberg — APPROVED** [review]

**Comment by voodoos** [comment]
Closing as part of https://github.com/ocaml/ocaml-lsp/pull/1383

**Pull request closed without merging**